### PR TITLE
`provider-ui` MUI framework upgrade

### PR DESCRIPTION
**Pull request by acald-creator**
**Notes for Reviewers**

This PR is for updating `provider-ui` to the latest
* Next.js 13
* React@18 and React-DOM@18
* MUI v5

One thing to note is that the Dockerfile has been changed to use `node:lts` which is `v18`, which is the same thing `ui` uses.

This also included some rearrangement of components, to reduce bloated and unused styles, and reduce the code being used in `_app.js`. This results in separate components called `Provider` and `Footer` along with custom styling for Provider.

The content via the drop-down menu has been changed to fit the width of the content, as before it was not.

The theme is one file with a few overrides, as there are only a few React components to create and use.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Review by Abhishek-kumar09 — COMMENTED** [review]
Did you compeleted the changes, solving the problems you were last discussing on the call around drop-downs?

**Comment by acald-creator** [comment]
Here is a screenshot of how it looks at the moment with the drop down menu issue. I'll need to make some more changes and this PR should be ready for review today.
provider-mui

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]
Looks Good to Me!
I just tested the UI locally, it works fine, just some style changes I would like to ask for:
old:
Screenshot 2023-01-26 at 10 29 46 PM

New:
Screenshot 2023-01-26 at 10 31 32 PM

---
Although I was not able to successfully `npm i`, the cypress installation was taking a lot of time, anyways I think that was my network issue.

File: provider-ui/.gitignore [hunk 1]
```diff
@@ -1,6 +1,18 @@
-# Cypress Tests Files
-cypress/videos
-cypress/screenshots
```
Comment (leecalcote) [inline provider-ui/.gitignore:3]: This seems like something we would want to ignore...

File: provider-ui/.eslintrc [hunk 2]
```diff
@@ -0,0 +1,62 @@
+module.exports = {
```
Comment (leecalcote) [inline provider-ui/.eslintrc:1]: This change is necessary for MUI v5 compatibility?

File: provider-ui/components/Footer.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import React, { Fragment } from 'react'
```
Comment (leecalcote) [inline provider-ui/components/Footer.js:1]: I might be missing something obvious. Why was this footer rewritten?

File: provider-ui/ui_dev_server.js [hunk 4]
```diff
@@ -1,39 +1,49 @@
...
+            } else {
+                if (req.url.startsWith('/provider')) {
+                    req.url = req.url.replace('/provider', '')
+                }
+                await handle(req, res, parsedUrl)
+            }
+        } catch (err) {
+            console.error('Error occurred handling', req.url, err)
+            res.statusCode = 500
+            res.end('internal server error')
```
Comment (leecalcote) [inline provider-ui/ui_dev_server.js:43]: Can we offer something more informative?

File: provider-ui/ui_dev_server.js [hunk 5]
```diff
@@ -1,39 +1,49 @@
...
+    createServer(async (req, res) => {
+        try {
+            // Be sure to pass `true` as the second argument to `url.parse`.
+            // This tells it to parse the query portion of the URL.
+            const parsedUrl = parse(req.url, true)
+            const { pathname, query } = parsedUrl
+
+            if (pathname.startsWith('/api')) {
+                proxy.web(req, res, {
+                    target: 'http://localhost:9081',
```
Comment (leecalcote) [inline provider-ui/ui_dev_server.js:32]: Why are hostname and port being defined as constants, but not referenced here?

File: provider-ui/cypress.config.js [hunk 6]
```diff
@@ -14,7 +14,7 @@ module.exports = defineConfig({
     setupNodeEvents(on, config) {
       return require('./cypress/plugins/index.js')(on, config)
     },
-    baseUrl: 'http://localhost:3001',
+    baseUrl: 'http://localhost:3001/provider',
```
Comment (leecalcote) [inline provider-ui/cypress.config.js:17]: Let's consider a separate improvement (separate issue) to dynamically attempt to use a higher port number in the case of port conflict on the contributor's system.

File: provider-ui/components/ProviderLayout.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import NoSsr from '@mui/material/NoSsr'
+
+export default function ProviderLayout({ children }) {
+  return (
+    <>
+      <NoSsr>
+        <div data-cy="root" style={{ padding: '170px 0px', textAlign: 'center' }}>
```
Comment (leecalcote) [inline provider-ui/components/ProviderLayout.js:9]: If we're rewriting, then let's not propagate bad practices of static styling, but instead move to flex.

File: provider-ui/ui_dev_server.js (hunk 4 above)
Comment (acald-creator) [reply provider-ui/ui_dev_server.js:46]: Let me think of something

File: provider-ui/components/Footer.js (hunk 3 above)
Comment (acald-creator) [reply provider-ui/components/Footer.js:1]: Footer wasn't rewritten, and it was moved to its own component. the `_app.js` was bloated with styles that weren't being used

File: provider-ui/.eslintrc (hunk 2 above)
Comment (acald-creator) [reply provider-ui/.eslintrc]: Not directly, sorry this was more to Next.js 13

File: provider-ui/.gitignore (hunk 1 above)
Comment (acald-creator) [reply provider-ui/.gitignore:3]: Fixed

File: provider-ui/components/Footer.js (hunk 3 above)
Comment (leecalcote) [reply provider-ui/components/Footer.js:1]: Ok, I see. Thanks for this. Makes much sense. 👍

**Review by leecalcote — DISMISSED** [review]
LGTM

**Review by Abhishek-kumar09 — APPROVED** [review]
Looks Good to Me

**Comment by Abhishek-kumar09** [comment]
Screenshot 2023-01-27 at 1 08 06 AM
eslint fixes pending

File: provider-ui/cypress.config.js (hunk 6 above)
Comment (acald-creator) [reply provider-ui/cypress.config.js]: I reverted the change, it was not intended to be this way.

File: provider-ui/components/Provider.js [hunk 8]
```diff
@@ -0,0 +1,260 @@
+import React, { Fragment, useEffect, useState } from "react";
```
Comment (Abhishek-kumar09) [inline provider-ui/components/Provider.js:1]: The tabwidth used in this file is 4 while in other it is 2. 
Can you please enforce 2, all around the project via eslintrc.

File: provider-ui/.eslintignore [hunk 9]
```diff
@@ -5,5 +5,7 @@ static
 package.json
 package-lock.json
 nodemon.json
-next.config.js
-cypress/
\ No newline at end of file
+lib/
+cypress/
+pages/_document.js
```
Comment (Abhishek-kumar09) [inline provider-ui/.eslintignore:10]: any specific reason for putting it in eslintignore?

Comment (Abhishek-kumar09) [reply provider-ui/.eslintignore:10]: `pages/_document.js`

**Review by Abhishek-kumar09 — APPROVED** [review]

**Pull request merged**